serverless/pages/ml-nlp-auto-scale.mdx

```diff
@@ -0,0 +1,128 @@
+
+There are two ways to enable autoscaling:
+
+- through APIs by enabling adaptive allocations
+- in Kibana by enabling adaptive resources
+
+
+Trained model autoscaling is available for Search, Observability, and Security projects on serverless deployments. However, these projects handle processing power differently, which impacts their costs and resource limits.
+
+Security and Observability projects are only charged for data ingestion and retention. They are not charged for processing power (vCU usage), which is used for more complex operations, like running advanced search models. For example, in Search projects, models such as ELSER require significant processing power to provide more accurate search results.
+
+## Enabling autoscaling through APIs - adaptive allocations
+
+Model allocations are independent units of work for NLP tasks.
+If you set the numbers of threads and allocations for a model manually, they remain constant even when not all the available resources are fully used or when the load on the model requires more resources.
+Instead of setting the number of allocations manually, you can enable adaptive allocations to set the number of allocations based on the load on the process.
+This can help you to manage performance and cost more easily.
+(Refer to the [pricing calculator](https://cloud.elastic.co/pricing) to learn more about the possible costs.)
+
+When adaptive allocations are enabled, the number of allocations of the model is set automatically based on the current load.
+When the load is high, a new model allocation is automatically created.
+When the load is low, a model allocation is automatically removed.
+You can explicitely set the minimum and maximum number of allocations; autoscaling will occur within these limits.
+
+<DocCallOut color="primary" title="Note">
+  If you set the minimum number of allocations to 1, you will be charged even if the system is not using those resources.
+</DocCallOut>
+
+You can enable adaptive allocations by using:
+
+- the create inference endpoint API for [ELSER](https://www.elastic.co/guide/en/elasticsearch/reference/master/infer-service-elser.html), [E5 and models uploaded through Eland](https://www.elastic.co/guide/en/elasticsearch/reference/master/infer-service-elasticsearch.html) that are used as inference services.
+- the [start trained model deployment](https://www.elastic.co/guide/en/elasticsearch/reference/master/start-trained-model-deployment.html) or [update trained model deployment](https://www.elastic.co/guide/en/elasticsearch/reference/master/update-trained-model-deployment.html) APIs for trained models that are deployed on machine learning nodes.
+
+If the new allocations fit on the current machine learning nodes, they are immediately started.
+If more resource capacity is needed for creating new model allocations, then your machine learning node will be scaled up if machine learning autoscaling is enabled to provide enough resources for the new allocation.
```

Comment:
is it worth clarifying that ml autoscaling is relevant for Cloud deployments? because at the beginning of the doc we only mentioned serverless.

Reply:
At the start of the document, I mainly aimed to clarify that, for serverless deployments, processing power, costs, and resource limits vary based on the project type (Search, Observability, or Security):

_"Trained model autoscaling is available for Search, Observability, and Security projects on serverless deployments. However, these projects handle processing power differently, impacting their costs and resource limits."_

However, I agree with you that it might imply autoscaling is only available on serverless deployments. To address this, rephrasing could help, like so:

_"Trained model autoscaling is available for both serverless and Cloud deployments. In serverless deployments, processing power is managed differently across Search, Observability, and Security projects, which impacts their costs and resource limits._